test/common/http/conn_manager_utility_test.cc

```diff
@@ -134,18 +134,21 @@ TEST_F(ConnectionManagerUtilityTest, InternalServiceForceTrace) {
     // Internal request, make traceable
     TestHeaderMapImpl headers{
         {"x-forwarded-for", "10.0.0.1"}, {"x-request-id", uuid}, {"x-envoy-force-trace", "true"}};
+    EXPECT_CALL(random_, uuid()).Times(0);
     EXPECT_CALL(runtime_.snapshot_, featureEnabled("tracing.global_enabled", 100, _))
         .WillOnce(Return(true));
     ConnectionManagerUtility::mutateRequestHeaders(headers, Protocol::Http2, connection_, config_,
                                                    route_config_, random_, runtime_, local_info_);
 
+    // Traceable (forced trace) variant of uuid
     EXPECT_EQ("f4dca0a9-12c7-a307-8002-969403baf480", headers.get_(Headers::get().RequestId));
```

Comment:
This particular UUID is a variant of UUID supplied by the client in the headers and not generated by the `MockRandomGenerator` (which are intentionally different).